pkg/block/s3/adapter.go

```diff
@@ -870,11 +870,13 @@ func (a *Adapter) ResolveNamespace(storageNamespace, key string, identifierType
 }
 
 func (a *Adapter) GetRegion(ctx context.Context, storageNamespace string) (string, error) {
-	bucket, found := strings.CutPrefix(storageNamespace, fmt.Sprintf("%s://", block.BlockstoreTypeS3))
+	namespace, found := strings.CutPrefix(storageNamespace, fmt.Sprintf("%s://", block.BlockstoreTypeS3))
```

Comment:
This seems somewhat fragile.  Proposal: I think you can use "net/url".Parse and then the bucket _is_ the Host field of the URL.  [Here's an example](https://go.dev/play/p/TKGisHOpbxL) of how this can work.